src/main/java/com/lambdaworks/redis/ReactiveCommandDispatcher.java

```diff
@@ -104,6 +104,12 @@ public void complete() {
                 super.complete();
 
                 if (getOutput() != null) {
+                    if (getOutput().hasError()) {
```

Comment:
Let's keep that block where it was before. The idea is to emit items which arrived before an error occurred and then terminate with a call to `onError`. 

Streaming output will still behave that way but non-streaming outputs would no longer emit items but terminate directly.